pages/dkp/kommander/1.4/clusters/tunnel-cli/index.md

```diff
@@ -0,0 +1,190 @@
+```
+
+## Connecting a remote cluster
+
+### Create a tunnel connector
+
+Create a tunnel connector on the management cluster for the remote cluster:
+```shell
+name_connector=sample-tunnel-connector
+
+cat > connector.yaml <<EOF
+apiVersion: kubetunnel.d2iq.io/v1alpha1
+kind: TunnelConnector
+metadata:
+  namespace: ${namespace}
+  name: ${name_connector}
+spec:
+  gatewayRef:
+    name: ${name_gateway}
+EOF
+
+cat connector.yaml
+
+kubectl apply -f connector.yaml
+```
+You can verify the gateway exists using:
+```shell
+kubectl get tunnelconnector -n ${namespace} ${name_connector}
+```
+
+Wait for the tunnel to reach `Listening` state and then export the agent manifest:
+
+```shell
+while [ "$(kubectl get tunnelconnector -n ${namespace} ${name_connector} -o jsonpath="{.status.state}")" != "Listening" ]
+do
+  sleep 5
+done
+
+name_manifest=$(kubectl get tunnelconnector -n ${namespace} ${name_connector} -o jsonpath="{.status.tunnelAgent.manifestsRef.name}")
+
+kubectl get secret -n ${namespace} ${name_manifest} -o jsonpath='{.data.manifests\.yaml}' | base64 -d > manifest.yaml
+```
+
+### Setup the managed cluster
+
+Copy the `manifest.yaml` file to the managed cluster and deploy the tunnel agent:
+
+```shell
+kubectl apply -f manifest.yaml
+```
+
+You can check the status of the created pods using:
+```shell
+kubectl get pods -n kubetunnel
+```
+
+There should be a `post-kubeconfig` job that reaches `Completed` state and a `tunnel-agent` deployment that stays in `Running` state.
+
+### Add the managed cluster into Kommander
+
+On the management cluster, wait for the tunnel to be connected by the tunnel agent:
+
+```shell
+while [ "$(kubectl get tunnelconnector -n ${namespace} ${name_connector} -o jsonpath="{.status.state}")" != "Connected" ]
+do
+  sleep 5
+done
+```
+
+Add the cluster into Kommander:
+```shell
+name_managed=private-cluster
+name_managed_display=${name_managed}
+
+cat > kommander.yaml <<EOF
+apiVersion: kommander.mesosphere.io/v1beta1
+kind: KommanderCluster
+metadata:
+  namespace: ${namespace}
+  name: ${name_managed}
+  annotations:
+    kommander.mesosphere.io/display-name: ${name_managed_display}
+spec:
+  clusterTunnelConnectorRef:
+    name: ${name_connector}
+EOF
+
+kubectl apply -f kommander.yaml
+```
+
+Wait for the managed cluster to join the Kommander cluster:
+```shell
+while [ "$(kubectl get kommandercluster -n ${namespace} ${name_managed} -o jsonpath='{.status.phase}')" != "Joined" ]
+do
+  sleep 5
+done
+
+name_kubefed=$(kubectl get kommandercluster -n ${namespace} ${name_managed} -o jsonpath="{.status.kubefedclusterRef.name}")
+while [ -z "${name_kubefed}" ]
+do
+  sleep 5
+  name_kubefed=$(kubectl get kommandercluster -n ${namespace} ${name_managed} -o jsonpath="{.status.kubefedclusterRef.name}")
+done
+
+kubectl wait --for=condition=ready --timeout=60s kubefedcluster -n kommander ${name_kubefed}
+
+kubectl get kubefedcluster -n kommander ${name_kubefed}
```

Comment:
It looks good to me, we could add the steps to validate the federated resources once attached as part of another PR.